### PR TITLE
Hconstr tweak debug and profile printing

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: kernel/hConstr.ml [hunk 1]
```diff
@@ -267,7 +271,10 @@ let nonrel_leaf tbl c = match kind c with
     Tbl.raw_find tbl h (fun c' -> eq_leaf c c')
   | _ -> None
 
+let steps = ref 0
```
Comment (ppedrot) [inline kernel/hConstr.ml:274]: Global references are not great, even for debugging purposes. Could you somehow store it directly in the local_env in of_constr?

Comment (SkySkimmer) [reply kernel/hConstr.ml]: I guess we can put the tbl there while we're at it, it's not evolving any differently than the binder tables which are already there

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by ppedrot** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**